src/main/java/com/zs/assignment10/dao/ProductDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+package com.zs.assignment10.dao;
+
+import com.zs.assignment10.exception.InternalServerException;
+import com.zs.assignment10.model.Product;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+public class ProductDaoImpl implements ProductDao {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProductDaoImpl.class);
+    private final ConnectionManager databaseConnection;
+    private String dbUrl;
+    private String userName;
+    private String password;
+
+    public ProductDaoImpl() {
+        this.databaseConnection = new ConnectionManager();
+        fetchProperties();
+    }
+
+    private void fetchProperties() {
+        Properties properties = new Properties();
+        FileInputStream fileInputStream;
+        try {
+            String homeDir = System.getProperty("user.dir");
+            fileInputStream = new FileInputStream(homeDir + "/src/main/resources/application.properties");
+            properties.load(fileInputStream);
+            dbUrl = properties.getProperty("DB_URL");
+            userName = properties.getProperty("USER");
+            password = properties.getProperty("PASSWORD");
+
+        } catch (IOException e) {
```

Comment:
In case of this exception, Application wont stop and continue to serve the menu driven program in case application.properties file fails to get loaded
Do we want to do that?